### PR TITLE
TRC-36-feat: add FeedItem component

### PR DESCRIPTION
**Pull request by willcary**
# TL/DR
This story creates a FeedItem component to be used for navigation.

FeedItem-Component

## Overview of Changes
`FeedItem.tsx`

- creates Props interface with `svg`, `label`, and `href`
- creates FeedItem component that accepts `svg`, `label`, and `href` as props to render an `<Icon />` component with a `label` to the right.

`FeedItem.scss`

- add scss file to style `FeedItem` component.

`FeedItem.stories.tsx`

- adds a default story with home icon, "Home" as the label prop, and `''` as href.

`FeedItem.test.tsx`

- Creates unit tests to ensure Icon and Label display in the document and verify that the href is passed correctly.

### CODE REVIEW

File: apps/web/src/components/FeedItem/FeedItem.stories.tsx
Comment (mmcbride2929) [inline apps/web/src/components/FeedItem/FeedItem.stories.tsx:1]: `const Template: ComponentStory<typeof FeedItem> = (args: Props) => (`

I could be wrong, so please take this with a grain of salt. But I do believe that you do not need to import Props, as your args are inferred as Props when you type your template as a `ComponentStory<typeof FeedItem>`

File: apps/web/src/components/FeedItem/FeedItem.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  line-height: 19px;
+  letter-spacing: 0em;
+  text-align: left;
+  color: #000000;
+  text-decoration: none;
+  cursor: pointer;
+}
+
+// Remove default underline style for anchor tag link (note: tried applying this style in .feedsbar-feeditem-label and elsewhere but it would not override the style below, even with !important)
+a:-webkit-any-link {
```
Comment (wijohnst) [inline apps/web/src/components/FeedItem/FeedItem.scss:22]: Have you tested this with other browsers? Targeting the `-webkit` selector may only override the text decoration for Safari?
 
So you'd also want to have:
```scss
a{
  text-decoration: none;
}
```
Also, I think that you're actually okay to target the `a` element vs. being more specific and removing the  underline in a class. We are using `Sass` so all of the CSS is modularized / compartmentalized. So this styling won't target any components that don't directly import the `FeedItem.scss` stylesheet or extend their stylesheet with it. Does that make sense?

What I'm saying is its okay to target top-level HTML elements with Sass / other post-CSS processors in ways that might be more problematic with regular / truly cascading CSS.

(If this isn't clear let's circle back to it in Slack because this is an important thing to understand.)

File: apps/web/src/components/FeedItem/FeedItem.stories.tsx
Comment (wijohnst) [reply apps/web/src/components/FeedItem/FeedItem.stories.tsx:1]: You are close to correct @mmcbride2929 , but the type inference for `args` isn't actually coming from `ComponentStory` directly. 

Consider this component:

```tsx
interface ButtonProps {
  label: string;
  primary?: boolean;
  onClick?: () => void;
}

export const Button = ({ label, primary, onClick }: ButtonProps) => (
  <button className={primary ? 'primary' : undefined} onClick={onClick}>
    {label}
  </button>
);
```

And this Story:

```tsx
import React from 'react';
import { Button } from './Button';

export const ButtonStory = (args: ButtonProps) => (
  <Button {...args} />
);

ButtonStory.args = {
  label: 'Button',
  primary: false,
  onClick: () => console.log('Button clicked'),
};
```

This is valid TypeScript and the `args` will be inferred correctly without having to import it directly. We need the additional type `ComponentStory` when we use the `Template.bind({})` syntax. 

To dig deeper on this see TypeScript's [Contextual Typing](https://www.typescriptlang.org/docs/handbook/type-inference.html#contextual-typing).

File: apps/web/src/components/FeedItem/FeedItem.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * @param {Object} props - The props object.
+ * @param {React.ReactElement} props.svg - The SVG icon to display
+ * @param {string} props.label - The text to show in the label element.
+ * @param {string} props.href - the URL to use in the anchor element.
+ * @returns {JSX.Element} A JSX element
+ */
+const FeedItem = ({ svg, label, href }: Props) => {
+  return (
+    <a href={href}>
```
Comment (wijohnst) [inline apps/web/src/components/FeedItem/FeedItem.tsx:22]: We can do a little more for `a11y` here by adding an `aria-label` to our link.
```suggestion
    <a href={href} aria-label={label}>
```

File: apps/web/src/components/FeedItem/FeedItem.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
```
Comment (wijohnst) [inline apps/web/src/components/FeedItem/FeedItem.tsx:1]: NIT
```suggestion
import * as React from 'react';
```

**Review by wijohnst — COMMENTED** [review]
Great work on this.

Left a few comments.

File: apps/web/src/components/FeedItem/FeedItem.stories.tsx
Comment (mmcbride2929) [reply apps/web/src/components/FeedItem/FeedItem.stories.tsx:1]: Makes sense!

File: apps/web/src/components/FeedItem/FeedItem.scss (hunk 1 above)
Comment (willcary) [reply apps/web/src/components/FeedItem/FeedItem.scss:22]: Good call! And that is good to know about the modular nature that Sass brings to the project. It does make sense to me and is super useful to be able to target top level elements without breaking styles in other components.

File: apps/web/src/components/FeedItem/FeedItem.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import { ComponentStory, ComponentMeta } from '@storybook/react';
+import { ReactComponent as HomeIcon } from '../../assets/homeIcon.svg';
+
+export default {
+  title: 'Components/FeedItem',
+  component: FeedItem,
+} as ComponentMeta<typeof FeedItem>;
+
+const Template: ComponentStory<typeof FeedItem> = (args) => (
+  <FeedItem {...args} />
```
Comment (wijohnst) [inline apps/web/src/components/FeedItem/FeedItem.stories.tsx:12]: Saw the big icon in the screen shot you posted to your PR description. Our `Icon` components will always grow to fill the box they are in. We can update how our story renders by modifying what is returned from out `Template`. Most valid JSX/TSX can be placed in the `return` statement of our `Template` function.

```suggestion
  <div style={{ height: "35px"}}>
    <FeedItem {...args} />
  </div>

```

Comment (wijohnst) [reply apps/web/src/components/FeedItem/FeedItem.stories.tsx:12]: This will render the `FeedItem` component in a container with a fixed height, which should result in something closed to the designs.

**Review by wijohnst — COMMENTED** [review]
Left a suggestion.

Please make sure you mark any suggestions that you correct as "resolved". Helps easily understand what's been updated without looking at the diff in detail.

**Review by wijohnst — APPROVED** [review]
Ship it!
